packages/api/src/derives/fees/estimateFee.ts

```diff
@@ -0,0 +1,56 @@
+import {ApiInterfaceRx} from '@cennznet/api/types';
+import Extrinsic from '@cennznet/types/extrinsic/Extrinsic';
+import {generateTransactionPayment} from '@cennznet/types/runtime/transaction-payment/TransactionPayment';
+import {AnyAssetId, IExtrinsic} from '@cennznet/types/types';
+import {drr} from '@polkadot/rpc-core/rxjs';
+import {TypeRegistry} from '@polkadot/types';
+import {RuntimeDispatchInfo} from '@polkadot/types/interfaces/rpc';
+import ExtrinsicEra from '@polkadot/types/primitive/Extrinsic/ExtrinsicEra';
+import BN from 'bn.js';
+import {combineLatest, Observable, of} from 'rxjs';
+import {catchError, first, map, switchMap} from 'rxjs/operators';
+
+export function estimateFee(api: ApiInterfaceRx) {
+  // We generate fake signature data here to ensure the estimated fee will correctly match the fee paid when the extrinsic is signed by a user.
+  // This is because fees are currently based on the byte length of the extrinsic
+  return (extrinsic: IExtrinsic, userFeeAssetId: AnyAssetId, maxPayment?: string): Observable<any> => {
+    return combineLatest([
+      api.rpc.state.getRuntimeVersion(),
+      api.rpc.chain.getBlockHash(),
+      api.rpc.chain.getBlockHash(0),
+      api.query.genericAsset.spendingAssetId(),
+    ]).pipe(
+      first(),
+      switchMap(
+        ([runtimeVersion, blockHash, genesisHash, networkFeeAssetId]): Observable<[RuntimeDispatchInfo, any]> => {
+          const registry = new TypeRegistry();
+          const era = new ExtrinsicEra(registry, {
+            current: 1,
+            period: 1,
+          });
+          const nonce = null;
+          const sender = '5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY';
```

Comment:
Could assign the address to `const ALICE = xxx; sender = ALICE;`, or alternatively add a comment at the end to reflect account name?

Reply:
Here it is not important to have ALICE's address or BOB's address.. doing this will create confusion as in, it is important to have ALICE's address as sender address.. Here it can be any address.

Reply:
maybe call it `fake_sender` to avoid confusion